### ISSUE TITLE
Decrease memory usage

### ISSUE THREAD

**Issue by merwok** [jeremymoreau/covid19mtl#17]
Notes from Jeremy:
Currently using ~460/512 MB RAM quota on heroku dyno

The way language localisations are currently handled (which is easy to maintain, but very suboptimal) and plots with animations are major causes.
Each translation takes up 50Mb of memory, reworking that could give us some breathing space.
The map figure for example uses a ton of memory because it creates a copy of the geojson for each frame of the animation instead of reusing it across days.

Note: We might not be able to implement all new features unless this is resolved as upgrading to a larger dyno with more memory is cost prohibitive.

**Comment by mschoettle** [comment]
@merwok Are you aware of any resource on the web regarding translating Dash apps?

**Comment by merwok** [comment]
It does not seem possible at the moment, see #21 and https://github.com/plotly/plotly.py/issues/2475 and the other tickets linked from there.

**Comment by mschoettle** [comment]
Thanks! That explains why I couldn't find anything.

I noticed that the layout is generated multiple times on app start. This is due to the language files all being imported and each calling `generate_layout`. I thought about only importing the required language file on page request, but then the layout is generated on the fly each time.

**Comment by merwok** [comment]
That would probably really slow down requests, and not reduce memory usage in the end (as the layouts would be loaded in memory eventually).

**Comment by mschoettle** [comment]
Good point. Might be easier to switch to plotly.js with Flask but not sure if it's worth it.

**Comment by jeremymoreau** [comment]
Yes, in retrospect I'd probably have made this with plotly.js (and just served it as a static site on e.g. netlify) if I were to do it again but the effort to convert everything doesn't seem worth it. For reference, this is now running on a VM with 2GB of RAM with usage at ~50% (but somewhat more RAM is required when deploying).
